ODLA/platforms/odla_popart/odla_popart.h

```diff
@@ -20,29 +20,42 @@
 #define ODLA_POPART_H_
 
 #include <ODLA/odla.h>
-
-#include <algorithm>
-#include <cassert>
-#include <chrono>
-#include <cmath>
-#include <cstddef>
-#include <functional>
-#include <memory>
-#include <numeric>
+#include <string>
+#include <vector>
+#include <atomic>
+#include <thread>
+#include <condition_variable>
 #include <popart/builder.hpp>
-#include <popart/dataflow.hpp>
-#include <popart/devicemanager.hpp>
-#include <popart/names.hpp>
-#include <popart/ndarraywrapper.hpp>
 #include <popart/session.hpp>
 #include <popart/sessionoptions.hpp>
-#include <popart/stepio.hpp>
 #include <popart/tensorinfo.hpp>
-#include <popart/voiddata.hpp>
-#include <random>
-#include <stdexcept>
-#include <string>
-#include <vector>
+
+#define g_comp _odla_computation::instance()
```

Comment:
can we use "static _odla_computation* &g_comp =  _odla_computation::instance_" ?

Reply:
We need the instance() function to do extra check of the thread id which is holding the computation object, like follows
`static _odla_computation* instance(bool hold_it = true){
      if(hold_it)
          instance_->hold(); 
      return instance_;
  }`